usecases/utilities_test.go

```diff
@@ -0,0 +1,377 @@
+			}
+		} else {
+			correctValue := strings.Contains(data, "<value>(1+2i)</value>")
+			if correctValue != true {
+				missingData = append(missingData, "Value")
+			}
+		}
+		correctUnit := strings.Contains(data, "<unit>testUnit</unit>")
+		if correctUnit != true {
+			missingData = append(missingData, "Unit")
+		}
+		correctVersion := strings.Contains(data, "<version>testVersion</version>")
+		if correctVersion != true {
+			missingData = append(missingData, "Version")
+		}
+		if len(missingData) != 0 {
+			return fmt.Errorf("missing data: %s", missingData)
+		}
+	} else {
+		missingData := []string{}
+		if expectedError == false {
+			correctValue := strings.Contains(data, `"value": 123`)
+			if correctValue != true {
+				missingData = append(missingData, "Value")
+			}
+		} else {
+			correctValue := strings.Contains(data, `"value": (1+2i)`)
+			if correctValue != true {
+				missingData = append(missingData, "Value")
+			}
+		}
+		correctUnit := strings.Contains(data, `"unit": "testUnit"`)
+		if correctUnit != true {
+			missingData = append(missingData, "Unit")
+		}
+		correctVersion := strings.Contains(data, `"version": "testVersion"`)
+		if correctVersion != true {
+			missingData = append(missingData, "Version")
+		}
+		if len(missingData) != 0 {
+			return fmt.Errorf("missing data: %s", missingData)
+		}
+	}
+	return nil
+}
+
+func TestPack(t *testing.T) {
+	params := []packParams{
+		{"application/xml", createTestForm(), false, "Best case, xml"},
+		{"application/json", createTestForm(), false, "Best case, json"},
+		{"application/xml", createBrokenTestForm(), true, "Bad case, xml"},
+		{"application/json", createBrokenTestForm(), true, "Bad case, json"},
+	}
+	for _, c := range params {
+		data, err := Pack(c.form, c.contentType)
+		if c.expectedError == false {
+			if err != nil {
+				t.Errorf("failed in testcase '%s' with error: %v", c.testCase, err)
+			}
+			err = assurePackData(data, c.contentType, c.expectedError)
+			if err != nil {
+				t.Errorf("error from assureData: %v", err)
+			}
+		} else {
+			if err == nil {
+				t.Errorf("expected error in testcase '%s', got none", c.testCase)
+			}
+		}
+	}
+}
+
+// This covers the case of it having a version but is not present in the form type map
+type testFormHasVersion struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+// This covers the case of it not having a version
+type testFormNoVersion struct {
+	Name string `json:"name"`
+}
+
+type unpackParams struct {
+	expectError bool
+	testCase    string
+	contentType string
+	setup       func() (data []byte, err error)
+}
+
+func TestUnpack(t *testing.T) {
+	testParams := []unpackParams{
+		//{expectError, testCase, contentType, setup()}
+		{false, "Best case, json", "text/plain", func() (data []byte, err error) {
+			var f forms.SignalA_v1a
+			f.NewForm()
+			data, err = json.Marshal(f)
+			return
+		}},
+		/* {false, "Best case, xml", "text/plain", func() (data []byte, err error) {
```

Comment:
Leftover or that broken test? If you want to keep this, leave a TODO note and explain why.